### PR TITLE
Forward more notifications to the WebContent process

### PR DESCRIPTION
**Pull request by pvollan**
#### 0c5c561cbd1e3ed66c4c55f791ac3710ffd7a551
<pre>
Forward more notifications to the WebContent process
<a href="https://bugs.webkit.org/show_bug.cgi?id=273532">https://bugs.webkit.org/show_bug.cgi?id=273532</a>
<a href="https://rdar.apple.com/127325980">rdar://127325980</a>

Reviewed by Sihui Liu and Chris Dumez.

Forward more notifications to the WebContent process from the UI process. These are
notifications we know are being observed in the WebContent process, and without this
change, their notification handler will not be invoked. If the notification handler is
not invoked, the WebContent process will not respond to the notification posts.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/Scripts/process-entitlements.sh:
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::WebProcessPool::registerNotificationObservers):
* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:

Canonical link: <a href="https://commits.webkit.org/278257@main">https://commits.webkit.org/278257@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/836e37be0562aa23aa15ce70039b8082cefd158b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/50009 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/29298 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/2292 "Built successfully") | [loading 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/53252 "Build is in progress. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running compile-webkit") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/686 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/35400 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/255 "Built successfully") | [loading 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/5/builds/53252 "Build is in progress. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running compile-webkit") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/52107 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/26885 "Passed tests") | [loading 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/43016 "Build is in progress. Recent messages:OS: Ventura (13.6.6), Xcode: 14.3; Running apply-patch; Checked out pull request") | [loading 🧪 api-wpe](https://ews-build.webkit.org/#/builders/5/builds/53252 "Build is in progress. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running compile-webkit") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/49862 "Passed tests") | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/8379 "Built successfully") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [loading 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/305 "Build is in progress. Recent messages:OS: Ventura (13.6.6), Xcode: 14.3; Skipping applying patch since patch_id isn't provided; Checked out pull request") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/54834 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/25103 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/223 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/48181 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/26361 "Built successfully") | [loading 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/43220 "Build is in progress. Recent messages:OS: Ventura (13.6.6), Xcode: 14.3; Running apply-patch; Checked out pull request") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/47220 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/10973 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/27223 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7213 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/26090 "Built successfully") | | |

### CODE REVIEW

**Review by cdumez — APPROVED** [review]

**Comment by pvollan** [comment]
Thanks for reviewing!

File: Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm [hunk 1]
```diff
@@ -716,8 +716,12 @@ static void logProcessPoolState(const WebProcessPool& pool)
         "com.apple.WebKit.showMemoryCache"_s,
         "com.apple.WebKit.showPaintOrderTree"_s,
         "com.apple.WebKit.showRenderTree"_s,
+        "com.apple.CFPreferences._domainsChangedExternally"_s,
```
Comment (szewai) [inline Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:719]: Do we need to update web process's sandbox profile to allow these for darwin-notification-post?

Comment (pvollan) [reply Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:719]: Ah, that is a good catch, fixed in latest patch!

Thanks for reviewing!

**Review by szewai — APPROVED** [review]

**Comment by cdumez** [comment]
> #### [dab3331](https://github.com/WebKit/WebKit/commit/dab33318cfd6a102a04e60ba8b726f4a8651c954)
> Forward more notifications to the WebContent process
> https://bugs.webkit.org/show_bug.cgi?id=273532
> [rdar://127325980](https://rdar.apple.com/127325980)
> 
> Reviewed by NOBODY (OOPS!).
> 
> Forward more notifications to the WebContent process from the UI process. This makes
> sure the WebContent process notification handlers are being called for these
> notifications.
> 
> * Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
> * Source/WebKit/Scripts/process-entitlements.sh:
> * Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
> (WebKit::WebProcessPool::registerNotificationObservers):
> * Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:
> [dab3331](https://github.com/WebKit/WebKit/commit/dab33318cfd6a102a04e60ba8b726f4a8651c954)

As discussed today, please fix the commit message to indicate what you're fixing, not just how you're fixing it.

**Comment by pvollan** [comment]
> > #### [dab3331](https://github.com/WebKit/WebKit/commit/dab33318cfd6a102a04e60ba8b726f4a8651c954)
> > Forward more notifications to the WebContent process
> > https://bugs.webkit.org/show_bug.cgi?id=273532
> > [rdar://127325980](https://rdar.apple.com/127325980)
> > Reviewed by NOBODY (OOPS!).
> > Forward more notifications to the WebContent process from the UI process. This makes
> > sure the WebContent process notification handlers are being called for these
> > notifications.
> > 
> > * Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
> > * Source/WebKit/Scripts/process-entitlements.sh:
> > * Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
> >   (WebKit::WebProcessPool::registerNotificationObservers):
> > * Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:
> >   [dab3331](https://github.com/WebKit/WebKit/commit/dab33318cfd6a102a04e60ba8b726f4a8651c954)
> 
> As discussed today, please fix the commit message to indicate what you're fixing, not just how you're fixing it.

Ah, yes, that's a good point, will fix!

Thanks for reviewing!

**Pull request merged**